src/services/refactors/ConvertToGetterAndSetter.ts

```diff
@@ -0,0 +1,109 @@
+/* @internal */
+namespace ts.refactor.ConvertGetterAndSetter {
+    const actionName = "Convert to getter and setter";
+    const description = Diagnostics.Convert_to_getter_and_setter.message;
+    registerRefactor(actionName, { getEditsForAction, getAvailableActions });
+
+    function getAvailableActions(context: RefactorContext): ApplicableRefactorInfo[] | undefined {
+        const { file, startPosition } = context;
+
+        const fieldInfo = getConvertibleFieldAtPosition(file, startPosition);
+        if (!fieldInfo) return undefined;
+
+        return [
+            {
+                name: actionName,
+                description,
+                actions: [
+                    {
+                        description,
+                        name: actionName
+                    }
+                ]
+            }
+        ];
+
+    }
+
+    function getEditsForAction(context: RefactorContext, _actionName: string): RefactorEditInfo | undefined {
+        const { file, startPosition } = context;
+        Debug.assertEqual(actionName, _actionName);
+
+        const fieldInfo = getConvertibleFieldAtPosition(file, startPosition);
+        if (!fieldInfo) return undefined;
+
+        const { fieldName, name, type, propertyDeclaration } = fieldInfo;
+        const changeTracker = textChanges.ChangeTracker.fromContext(context);
+
+        const getter = createGetter(fieldName, name, type);
+        const setter = createSetter(fieldName, name, type);
+
+        changeTracker.insertNodeAfter(file, propertyDeclaration, getter);
+        changeTracker.insertNodeAfter(file, propertyDeclaration, setter);
+
+        return {
+            edits: changeTracker.getChanges(),
+            renameFilename: undefined,
+            renameLocation: undefined,
+        };
+    }
+
+    interface Info { fieldName: string; name: string; type: TypeNode; propertyDeclaration: PropertyDeclaration; }
+    function getConvertibleFieldAtPosition(file: SourceFile, startPosition: number): Info | undefined {
+        const node = getTokenAtPosition(file, startPosition, /*includeJsDocComment*/ false);
+        const propertyDeclaration = findAncestor(node.parent, isPropertyDeclaration);
+        if (!(propertyDeclaration && propertyDeclaration.name.getText().charAt(0) === "_" && hasModifier(propertyDeclaration, ModifierFlags.Private))) return undefined;

```

Comment:
Give a short comment explaining why you want to do this check.

Reply:
Also, does this work for computed properties? You probably want to make sure that your `propertyDeclaration.name` is an identifier. That way you could also use `propertyDeclaration.name.text` after you use a type-guard.

Reply:
I don't like that you have to write `private _foo: number;` to get this refactor. Wouldn't it make more sense to trigger on `public foo: number;` and convert it to `private _foo: number; public get foo() {} public set foo() {}`?

Reply:
updated